zerver/lib/send_email.py

```diff
@@ -110,15 +115,24 @@ def send_email(template_prefix: str, to_user_ids: Optional[List[int]]=None,
                to_emails: Optional[List[str]]=None, from_name: Optional[str]=None,
                from_address: Optional[str]=None, reply_to_email: Optional[str]=None,
                language: Optional[str]=None, context: Dict[str, Any]={}) -> None:
-    mail = build_email(template_prefix, to_user_ids=to_user_ids, to_emails=to_emails,
-                       from_name=from_name, from_address=from_address,
-                       reply_to_email=reply_to_email, language=language, context=context)
+    # Callers should pass exactly one of to_user_id and to_email.
+    assert (to_user_ids is None) ^ (to_emails is None)
+    if to_user_ids is None:
+        emails = [build_email(template_prefix, to_emails=to_emails, from_name=from_name,
+                              from_address=from_address, reply_to_email=reply_to_email,
+                              language=language, context=context)]
+    else:
+        emails = build_emails_by_user_ids(template_prefix, to_user_ids=to_user_ids, from_name=from_name,
+                                          from_address=from_address, reply_to_email=reply_to_email,
+                                          language=language, context=context)
```

Comment:
I don't like this split since it involves copying a bunch of arguments around; can we have the conditional live in `build_email` instead?